osprey_flask_app/routes.py

```diff
@@ -68,6 +80,48 @@ def models_route():
     return Response(f"Available climate models:<br><br>{model_list}", status=201)
 
 
+@osprey.route("/progress/<job_id>")
+def progress_route(job_id):
+    def get_percent_and_timestamp(date_format, end, total_days):
+        address = ("osprey-dev", os.environ.get("LISTENER_PORT", 5005))
+        try:
+            with Client(address) as conn:
+                message = conn.recv_bytes().decode("utf-8")
+            timestamp = re.search(r"\d{4}-\d{2}-\d{2}", message)[0]
+            timestamp = datetime.strptime(timestamp, date_format)
+            days_left = (end - timestamp).days
+            percent = 100 - int(
+                days_left * 90 / total_days
+            )  # Start convolution progress at 10 percent
+        except ConnectionRefusedError:  # Listener thread for convolution process has not been created. Still in parameters process.
+            percent = 9
+            timestamp = ""
+        return (percent, timestamp)
+
+    def generate():
+        start_date_format = "%Y-%m-%d-%H"
+        stop_date_format = "%Y-%m-%d"
+        start = datetime.strptime(dates[job_id][0], start_date_format)
+        end = datetime.strptime(dates[job_id][1], stop_date_format)
+        total_days = (end - start).days
+
+        (percent, timestamp) = get_percent_and_timestamp(
```

Comment:
I don't think that we have a better alternative at this point, but we should note here (in a comment, maybe) that this uses blocking I/O. I don't think that there's any timeout or anything, and I'm not sure if the behaviour is predictable if the other side of the connection quietly goes away.